contracts/modules/licensing/LicensingModule.sol

```diff
@@ -32,57 +34,67 @@ import { BaseModule } from "../BaseModule.sol";
 /// - Linking IP to its parent
 /// - Verifying linking parameters
 /// - Verifying policy parameters
-contract LicensingModule is AccessControlled, ILicensingModule, BaseModule, ReentrancyGuard {
+contract LicensingModule is AccessControlled, ILicensingModule, BaseModule, ReentrancyGuardUpgradeable, GovernableUpgradeable, UUPSUpgradeable {
     using ERC165Checker for address;
     using IPAccountChecker for IIPAccountRegistry;
     using EnumerableSet for EnumerableSet.UintSet;
     using EnumerableSet for EnumerableSet.AddressSet;
     using Licensing for *;
     using Strings for *;
 
+    /// @notice Storage struct for the LicensingModule
+    /// @param registeredFrameworkManagers Mapping of registered policy framework managers
+    /// @param hashedPolicies Mapping of policy data to policy id
+    /// @param policies Mapping of policy id to policy data (hashed)
+    /// @param totalPolicies Total amount of distinct licensing policies in LicenseRegistry
+    /// @param policySetups Internal mapping to track if a policy was set by linking or minting, and the index of the policy in the
+    /// ipId policy set. Policies can't be removed, but they can be deactivated by setting active to false.
+    /// @param policiesPerIpId the set of policy ids attached to the given ipId
+    /// @param ipIdParents Mapping of parent policy ids for the given ipId
+    /// @param ipRights Mapping of policy aggregator data for the given ipId in a framework
+    /// @custom:storage-location erc7201:story-protocol.LicensingModule
+    struct LicensingModuleStorage {
```

Comment:
you shouldn't mix code optimization with upgrade PR.  I would expect a small PR just for upgrade.

Reply:
Or it is a prerequisite of the upgrade?  Please clarify.

Reply:
Not sure what would be the optimization here. ERC7201 requires that the storage of a contract is in a random but deterministic storage slot (to avoid collisions between contracts, upgraded implementations, etc)

For that, we must put the storage in a struct (so it all has the same storage slot as "starting point")

The immutable variables don't occupy storage slots (the compiler inlines the values), so they don't need to be there.